### ISSUE TITLE
implement GEORADIUS

### ISSUE THREAD

**Issue by romange** [dragonflydb/dragonfly#1878]
see https://redis.io/commands/georadius/ for details

**Comment by yoelsherwin** [comment]
Hi @romange , I would like to give this a go. Were you thinking of implementing the georadius in its generic way like in redis? Or starting by something simple and making it generic in the future?

**Comment by romange** [comment]
Hey @yoelsherwin ! What do you mean by a generic way like in Redis? if you ask whether we must support all the options from the start then no, we do not need to. as long as your code brings us closer, you can choose the scope of your PR.

**Comment by yoelsherwin** [comment]
I mean that in Redis, geoRadius (along with other commands) is basically just calling the `georadiusGeneric` with different parameters. I was not sure if that the right way to implement it (generalizing where we don't need it yet), but since the end goal is known maybe this is not such a bad idea. Would love to hear your thoughts :)

**Comment by romange** [comment]
I have not looked into the code. 1. we can not use `georadiusGeneric` unfortunately, because it uses `client*` which we do not have. Therefore it needs to be reimplemented. I would not be bothered at this point with "generic" implementation fitting all the options/usecases, and I would choose a specific option set at API level and implement only it at first. Without STORE, for example. Only for `M`, only `WITHCOORD` etc.

**Comment by romange** [comment]
basically an MVP which can gradually grow

**Comment by yoelsherwin** [comment]
Yeah, I was aware that we need to reimplement it, just wasn't sure at what scope. Anyway, got the answer I was looking for :) Thanks!
Will probably get to it around next week

**Comment by yoelsherwin** [comment]
hi @romange , I started working on this but I won't be able to finish this soon due to personal reasons. Sorry. hope I could contribute again soon 🙏

**Comment by romange** [comment]
Hi @yoelsherwin hope everything is all right and thanks again for your contributions!

**Comment by azuredream** [comment]
Hi @romange ,
According to https://redis.io/docs/latest/commands/georadius/, GEORADIUS can be replaced by GEOSEARCH with BYRADIUS option. 
If you still want this, please assign it to me. 
I think I just need to make some changes based on the impl of GEOSEARCH.

**Comment by romange** [comment]
Yes, we must have a single implementation and probably have GEORADIUS wrapping GEOSEARCH.

**Comment by azuredream** [comment]
Hi @romange , PR created. 
#3756
